rpcs3/Emu/RSX/Common/FragmentProgramDecompiler.cpp

```diff
@@ -373,14 +371,22 @@ std::string FragmentProgramDecompiler::Format(const std::string& code, bool igno
 
 std::string FragmentProgramDecompiler::GetRawCond()
 {
-	static const char f[4] = { 'x', 'y', 'z', 'w' };
+	static constexpr std::string_view f = "xyzw";
 
 	std::string swizzle, cond;
+	swizzle.reserve(5);
 	swizzle += f[src0.cond_swizzle_x];
 	swizzle += f[src0.cond_swizzle_y];
 	swizzle += f[src0.cond_swizzle_z];
 	swizzle += f[src0.cond_swizzle_w];
-	swizzle = swizzle == "xyzw" ? "" : "." + swizzle;
+	if (swizzle != "xyzw"sv)
+	{
+		swizzle.insert(0, ".");
```

Comment:
You know you can avoid this insert business by just having the swizzle start with a "." and avoid shifting bytes unnecessarily if you're really trying to improve this. I don't expect any performance differences from these changes on modern CPUs and compilers either way. Same goes for all the other places.

Reply:
You see, I know barely nothing about the logic here. So I tried to optimize existing. Do you mean, that swizzle strings can always begin with ```.```? So if I write ```reserve(5)``` and ```[0] = '.'``` and shift everything after, I'l break nothing? If so, I can optimize it a bit more.

Reply:
It's not about the code's logic, it about the string's usage which is what you're doing already anyway. The strings ends up one of two ways:
1. insert(".") and append a mask
2. clear()
So the string can just start with a "." anyway and if it matches ".xyzw" (with a dot this time) then it gets cleared else it gets appended as-is without insert.

Reply:
Is this a correct variant?
```
std::string swizzle;
swizzle.reserve(5);
swizzle += '.';
swizzle += f[src.swizzle_x];
swizzle += f[src.swizzle_y];
swizzle += f[src.swizzle_z];
swizzle += f[src.swizzle_w];

if (swizzle == ".xxxx") swizzle = ".x";
else if (swizzle == ".yyyy") swizzle = ".y";
else if (swizzle == ".zzzz") swizzle = ".z";
else if (swizzle == ".wwww") swizzle = ".w";

if (swizzle != ".xyzw"sv)
{
	ret += swizzle;
}
```
And another one
```
std::string swizzle, cond;
swizzle.reserve(5);
swizzle += '.';
swizzle += f[src0.cond_swizzle_x];
swizzle += f[src0.cond_swizzle_y];
swizzle += f[src0.cond_swizzle_z];
swizzle += f[src0.cond_swizzle_w];

if (swizzle == ".xxxx") swizzle = ".x";
else if (swizzle == ".yyyy") swizzle = ".y";
else if (swizzle == ".zzzz") swizzle = ".z";
else if (swizzle == ".wwww") swizzle = ".w";

if (swizzle == ".xyzw"sv)
{
	swizzle.clear();
}
```

Reply:
Sure, that works

Reply:
Ok, will update soon

Reply:
@kd-11, I've made a change as you suggested. Could you please check it.